packages/compiler_dart/lib/src/body.dart

```diff
@@ -232,11 +232,33 @@ class DartExpressionVisitor extends ExpressionVisitor<List<dart.Code>> {
         ],
         dart.Code('break ${_label(node.scopeId)};'),
       ];
+
   @override
   List<dart.Code> visitContinueExpression(ContinueExpression node) => [
         dart.Code('continue ${_label(node.scopeId)};'),
       ];
 
+  @override
+  List<dart.Code> visitAssignmentExpression(AssignmentExpression node) => [
+        ...node.right.accept(this),
+        node.left.identifier
+            .maybeMap(
+              property: (property) => dart.refer(
```

Comment:
This doesn't work for properties in a trait or class, but we can wait for #15's changes to properties in the HIR.